### DISCUSSION TITLE
Api filter(search) branches by name

### DISCUSSION THREAD

**Discussion by mariangemarcano** [community/community#24512] [Apps, API and Webhooks]
<p>Is it possible, using gihub api v3, get a list of branches in a repository that matches a name (or part of the name), looking for a branch search support?</p>
<p>By looking at the docs I can get the branch list or get info for a single specific branch, but I can’t find a search support? </p>
<p>-  <a href="https://developer.github.com/v3/repos/branches/" rel="nofollow noopener">https://developer.github.com/v3/repos/branches/ </a></p>
<p>- <a href="https://developer.github.com/v3/search/" rel="nofollow noopener">https://developer.github.com/v3/search/</a></p>

**Comment by lee-dohm** [comment]
<p>Good question.</p>
<p>If I’m understanding what you’re describing correctly, you’ll have to get the list of all branches and then do the matching or filtering yourself. There isn’t a “search for a branch” feature in the GitHub API.</p>
<p>I hope that helps!</p>

**Comment by gbidsilva** [comment]
<p>Is this the same behavior still?<br>
As I went through the documentation, I could not find a proper API to perform search on Branches and Tags.<br>
Is there an API available for searching branches and tags now?</p>

**Comment by lee-dohm** [comment, marked as answer]
<p>That’s correct. There is not a “search for a branch” feature in the GitHub API.</p>

**Comment by ankittiwari-harness** [comment]
Hi Team, 

Is it planned for future. Gitlab has regex support in list branch API to support this out of the box. https://docs.gitlab.com/ee/api/branches.html

<img width="997" alt="image" src="https://github.com/community/community/assets/110376355/a2910eb8-a365-4f75-8f07-2d53a99b749b">


Are we tracking this issue somewhere or it's not planned to be picked up anytime soon.

**Comment by JP250552** [comment]
It would be great to have the ability to filter by branch name. 
We have a monorepo with over 400 branches (whether we need that many is a separate concern) and the API has a max of 100 per page. 
We have a script that queries for `release/*` branches, but need to page through several requests because they don't all show up in the first 100.

**Reply by JP250552** [reply]
I do see there is a `protected` param for returning only protected branches that actually will work for my use case.
